src/kirin/ir/attrs/py.py

```diff
@@ -51,3 +55,19 @@ def print_impl(self, printer: Printer) -> None:
 
     def unwrap(self) -> T:
         return self.data
+
+    def serialize(self, serializer: "Serializer") -> dict[str, Any]:
```

Comment:
what if I have the following attribute

```python
class MyAttr(Data[T]):
     data: T
     type: SomeCustomType
```

this serialization target format cannot recover the original attribute information, we can solve this by also serializing the name of an attribute.